src/foam/u2/grid/testdata/HeroViews.js

```diff
@@ -0,0 +1,213 @@
+foam.CLASS({
+  name: 'HeroCellView',
+  extends: 'foam.u2.Element', 
+  
+  requires: ['Team', 'Hero', 'foam.u2.Element'],
+  
+  imports: [
+  ], 
+  
+  properties: [
+    "of",
+    {
+        name: 'data',
+        postSet: function(old ,nu){
+            nu && nu.sub && nu.sub(this.onDataUpdate);
+        }, 
+    },
+    {
+        name: 'cellView',
+        factory: function(){
+            return this.Element.create(null, this);
+        }
+    }
+    
+    ],
+  
+  methods: [
+    function initE(){
+        this.add(this.cellView$);
+        this.attrs({ draggable: 'true' });
+        this.on('dragstart', this.onDragStart); 
+    },
+    
+    function init(){
+        this.onDataUpdate();
+    },
+    
+    function makeCellView(){
+        var div = foam.u2.Element.create("div");
+        if (this.data){
+            div.add(foam.u2.Element.create("p").add(this.data.name));
+            div.start("small").add(this.data.lastSeenAlive && this.data.lastSeenAlive.toLocaleDateString()).end("small"); 
+        }
+        //return div;
+        this.cellView = div; 
+    }
+    
+  ],
+  
+    
+    listeners: [
+        {
+            name: "onDataUpdate",
+            code: function(){
+              this.makeCellView();
+            }
+        },
+        {
+          name: 'onDragStart',
+          code: function(e) {
+          e.dataTransfer.setData('application/x-foam-obj-id', this.data.id);
+          e.stopPropagation();
+          }
+        },
+      ],
+  
+});
+  
+  
+foam.CLASS({
+  name: 'NewHeroWrapperView',
+  extends: 'foam.u2.Element', 
+  
+  requires: [
+    'Team',
+    'Hero',
+    'foam.dao.AbstractSink'
+    ],
+  
+  imports: [
+    
+    'HeroDAO',
+    'TeamDAO',
+  ], 
+  
+  properties: [
+    "of",
+
+    {
+        name: 'wrapperView',
+        factory: function(){
+            return foam.u2.Element.create();
+        }
+    },
+    "status",
+    "organizationId",
+    "cell", 
+    ],
+  
+  methods: [
+    function initE(){
+        //this.cssClass(this.myCls()).
+        this.
+          style({
+            'border': '2px solid red', 
+            'height':'100%',
+            'width':'100%',
+            'display':'inline-block',
+            }).
+          start('div', null, this.content$).
+            cssClass(this.myCls('content')).
+          end().
+          add(this.wrapperView$);
+          
+        this.on('dragenter', this.onDragOver).
+          on('dragover', this.onDragOver).
+          on('drop', this.onDrop); 
+    },
+    
+    function makeWrapper(){
+        var div = foam.u2.Element.create("div");
+        
+        div.start(this.NEW, {data: this}).end() ;
+          
+        this.wrapperView = div; 
+    }, 
+    
+    function init(){
+      //this.organizationId$ = this.cell.rowMatchId$;
+      this.organizationId$ = this.cell.rowMatchId$;
+      this.status$ = this.cell.colMatch$; 
+        this.onDAOUpdate();
+    },
+    
```

Comment:
Generally, formatting is inconsistent with {'s not lining up and multiple blank lines.